python/ray/data/_internal/datasource/parquet_datasink.py

```diff
@@ -149,6 +151,14 @@ def __init__(
             if "compression" in open_stream_args:
                 self.arrow_parquet_args["compression"] = open_stream_args["compression"]
 
+        if ("partitioning_flavor" in self.arrow_parquet_args) or (
+            self.arrow_parquet_args_fn is not None
+            and "partitioning_flavor" in self.arrow_parquet_args_fn()
+        ):
+            assert (
```

Comment:
Nit: I think `ValueError` is more appropriate here because this is an error with user-provided values rather an internal correctness